.circleci/getChangelogs.sh

```diff
@@ -32,23 +32,46 @@ function cloneRepository() {
     old_version=$2
     new_version=$3
     repository_url=$4
-    echo "[OK] $project_name($repository_url) : $old_version -> $new_version"
+    echo "[OK] checkout: $project_name($repository_url)"
     git clone "$repository_url" "$project_name" --quiet
 }
 
+function getOldDependencyVersion(){
+    old_version=$(getDependencyVersion $OLD_LOCK_FILE $1)
+    if [ "$old_version" == "" ]; then
+        # this means this dependency is new as there is no old_version as per the requirements file
+        # therefore we fetch the oldest tag from the git remote
+        old_version=$(git ls-remote --tags $2 | sort -t '/' -k 3 -V | awk '{print $2}' | sed 's/refs\/tags\///g' | head -n 1)
+    fi
+    echo "$old_version"
+}
+
+function getDependencies() {
+    echo $(yq r $1 dependencies -j | jq ".[].name" | sed "s/\"//g")
+}
+
+function getDependencyVersion() {
+    echo $(yq r $1 dependencies -j | jq ".[] | select(.name==\"$2\").version" | sed "s/\"//g")
+}
+
+function getDependencyRepoUrl() {
+    echo $(yq r $1 dependencies -j | jq ".[] | select(.name==\"$2\").git" | sed "s/\"//g")
+}
+
 rm -rf ./changelogs
 mkdir changelogs
 prepareChangelogMarkdown
 
-LATEST_TAG="stable"
+LATEST_TAG="latest"
```

Comment:
this caused the latest release to have too many tags in the changelog. 
latest is the correct tag to use